src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -12,6 +12,55 @@ namespace System.Text.Json.Tests
 {
     public static partial class Utf8JsonReaderTests
     {
+        // Test string, Argument to DateTime(Offset).Parse(Exact)
+        public static IEnumerable<object[]> ValidISO8601Tests()
+        {
+            yield return new object[] { "\"0997-07-16\"", "0997-07-16" };
+            yield return new object[] { "\"1997-07-16\"", "1997-07-16" };
+            yield return new object[] { "\"1997-07-16T19:20\"", "1997-07-16T19:20" };
+            yield return new object[] { "\"1997-07-16T19:20:30\"", "1997-07-16T19:20:30" };
+            yield return new object[] { "\"1997-07-16T19:20:30.45\"", "1997-07-16T19:20:30.45" };
+            yield return new object[] { "\"1997-07-16T19:20:30.4555555\"", "1997-07-16T19:20:30.4555555" };
+
+            // Skip test T24:00 till #35830 is fixed.
+            // yield return new object[] { "\"1997-07-16T24:00\"", "1997-07-16T24:00" };
+
+            // Test fraction rounding.
+            yield return new object[] { "\"1997-07-16T19:20:30.45555554\"", "1997-07-16T19:20:30.45555554" };
+            // We expect the parser to truncate. `DateTime(Offset).Parse` will round up to 7dp in this case,
+            // so we pass a string representing the Datetime(Offset) we expect to the `Parse` method.
+            yield return new object[] { "\"1997-07-16T19:20:30.45555555\"", "1997-07-16T19:20:30.4555555" };
```

Comment:
Extra digits for the fractional part might be allowed (up to 9), but what about beyond that? Also, what about if we had non-digits after?

Reply:
We parse strings with any amount of fractional digits (> 0), but truncate after 7 decimal places.
If we had non-digits after, we'd return `false`/throw a `FormatException`.
Please see test cases for details.